app/controllers/participants_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class ParticipantsController < ApplicationController
+  before_action :set_participant, only: %i[show edit update destroy]
```

Comment:
if we use `cancancan` we can go for this and automatically only pass it, if the user has access.

```rb
  load_and_authorize_resource :unit
  load_and_authorize_resource :participant, through: :participant, except: [:create]
```